internal/cephfs/volume.go

```diff
@@ -144,7 +150,13 @@ func createVolume(ctx context.Context, volOptions *volumeOptions, cr *util.Crede
 // subvolume. If the command is not available as a fallback it will use
 // CreateVolume to resize the subvolume.
 func resizeVolume(ctx context.Context, volOptions *volumeOptions, cr *util.Credentials, volID volumeID, bytesQuota int64) error {
-	if supported, ok := resizeSupportedList[volOptions.ClusterID]; supported || !ok {
+	// verify if corresponding ClusterID entry is present in the map,
+	// and if not, initialize with default values(false).
+	if _, ok := clusterAdditionalInfo[volOptions.ClusterID]; !ok {
+		clusterAdditionalInfo[volOptions.ClusterID] = &additionalClusterInfo{}
+	}
+
+	if supported := clusterAdditionalInfo[volOptions.ClusterID].resizeSupported; supported {
```

Comment:
This seems incorrect.this will be subvolume resize is called only if the resizeSupported is set to true. we  also need to try to resize for the first time to check whether resize is supported or not

Reply:
Thanks, I'll make the changes.

Reply:
I had a slight doubt regarding this, were we checking whether resize is supported beforehand in https://github.com/ceph/ceph-csi/blob/master/internal/cephfs/volume.go#L147 ?

Reply:
I had a doubt regarding this, is resizevolume skipped for the first run since resize supported is set to false? https://github.com/ceph/ceph-csi/blob/master/internal/cephfs/volume.go#L147

Reply:
we have double check here, `if supported, ok := resizeSupportedList[volOptions.ClusterID]; supported || !ok {` either its supported or key not present